src/shared/dissect-image.h

```diff
@@ -26,6 +26,7 @@ struct DissectedPartition {
         char *fstype;
         char *node;
         char *label;
+        int decrypted_node_fd;
```

Comment:
hmm, instead of doing this only if decryption is in place, I'd always pin the device here once we know which one to mount. i.e. call this `mount_node_fd` or so, then always open the block device one we figured it out both if encrypted or if not, and then always mount through /proc/self/fd/.

It think that's much prettier since it adds less special casing to the encryption code path